src/view.rs

```diff
@@ -189,8 +189,10 @@ pub fn view_grammar(m: &Model, coord: Coordinate) -> Html {
             }
             Kind::Input(value) => {
                 let is_active = m.active_cell.clone() == Some(coord.clone());
-                let first_select_cell = m.first_select_cell.clone() == Some(coord.clone());
-                let last_select_cell = m.last_select_cell.clone() == Some(coord.clone());
+                let is_select = m.last_select_cell.clone() == Some(coord.clone());
+                let mut first_select_cell  = m.first_select_cell.clone();
+                let mut last_select_cell  = m.last_select_cell.clone();             
```

Comment:
instead of defining these up here and passing them into `view_input_grammar`, why don't you retrieve them from the model inside `view_input_grammar`?